services/shotgrid_common/ayon_shotgrid_hub/match_ayon_hierarchy_in_shotgrid.py

```diff
@@ -30,95 +42,128 @@ def match_ayon_hierarchy_in_shotgrid(entity_hub, sg_project, sg_session, project
+
+                if sg_ay_dict["data"][CUST_FIELD_CODE_ID] != ay_entity.id:
+                    logging.error("Shotgrid record for AYON id does not match...")
+                    try:
+                        sg_session.update(
+                            sg_ay_dict["attribs"][SHOTGRID_TYPE_ATTRIB],
+                            sg_ay_dict["attribs"][SHOTGRID_ID_ATTRIB],
+                            {
+                                CUST_FIELD_CODE_ID: "",
+                                CUST_FIELD_CODE_SYNC: "Failed"
+                            }
+                        )
+                    except Exception as e:
+                        log_traceback(e)
+                        ay_project_sync_status = "Failed"
+                else:
+                    # Update SG entity custom attributes with AYON data
+                    data_to_update = get_sg_custom_attributes_data(
+                        sg_session,
+                        ay_entity.attribs.to_dict(),
+                        sg_entity_type,
+                        custom_attribs_map
+                    )
+                    if data_to_update:
+                        logging.info("Syncing custom attributes on entity.")
+                        sg_session.update(
+                            sg_entity_type,
+                            sg_entity_id,
+                            data_to_update
+                        )
+
+            if sg_ay_dict is None:
                 sg_parent_entity = sg_session.find_one(
-                    ay_parent_entity["shotgridType"],
-                    filters=[["id", "is", ay_parent_entity["shotgridId"]]]
+                    sg_ay_parent_entity["attribs"][SHOTGRID_TYPE_ATTRIB],
+                    filters=[["id", "is", sg_ay_parent_entity["attribs"][SHOTGRID_ID_ATTRIB]]]
                 )
-                sg_entity = _create_new_entity(
+                sg_ay_dict = _create_new_entity(
                     ay_entity,
                     sg_session,
                     sg_project,
                     sg_parent_entity,
-                    project_code_field
+                    sg_enabled_entities,
+                    project_code_field,
+                    custom_attribs_map,
                 )
-                sg_entity_id = sg_entity["shotgridId"]
-                sg_entities_by_id[sg_entity_id] = sg_entity
-                sg_entities_by_parent_id[sg_parent_entity["id"]].append(sg_entity)
+                sg_entity_id = sg_ay_dict["attribs"][SHOTGRID_ID_ATTRIB]
+                sg_ay_dicts[sg_entity_id] = sg_ay_dict
+                sg_ay_dicts_parents[sg_parent_entity["id"]].append(sg_ay_dict)
 
             ay_entity.attribs.set(
                 SHOTGRID_ID_ATTRIB,
                 sg_entity_id
             )
             ay_entity.attribs.set(
                 SHOTGRID_TYPE_ATTRIB,
-                sg_entity["type"]
+                sg_ay_dict["type"]
```

Comment:
```suggestion
                sg_ay_dict["attribs"][SHOTGRID_TYPE_ATTRIB]
```
This should be like this, otherwise it's filling up `folder` as the SG entity type